app/(header-default)/cached-methods.ts

```diff
@@ -0,0 +1,51 @@
+import { redirect } from 'next/navigation';
+import { cache } from 'react';
+import { extractSirenOrSiretSlugFromUrl } from '#utils/helpers';
+import { getEtablissementWithUniteLegaleFromSlug } from '../../models/core/etablissement';
+import {
+  NotASirenError,
+  NotASiretError,
+  SirenNotFoundError,
+  SiretNotFoundError,
+} from '../../models/core/types';
+import { getUniteLegaleFromSlug } from '../../models/core/unite-legale';
+
+const handleException = (e: any, slug: string) => {
+  if (e instanceof NotASirenError || e instanceof NotASiretError) {
+    redirect('/404');
+  }
+  if (e instanceof SirenNotFoundError || e instanceof SiretNotFoundError) {
+    redirect('/erreur/introuvable/' + slug);
+  }
+};
+
+/**
+ *  Call this function to rely on react cache when using an unite legale
+ */
+export const cachedGetUniteLegale = cache(
+  async (slug: string, isBot: boolean, page = 0) => {
+    const sirenSlug = extractSirenOrSiretSlugFromUrl(slug);
+    try {
+      const uniteLegale = await getUniteLegaleFromSlug(sirenSlug, {
+        isBot,
+        page,
+      });
+      return uniteLegale;
+    } catch (e) {
+      handleException(e, sirenSlug);
+      throw e;
```

Comment:
So we redirect and throw at the same time. Not sure to know how this is handled by next. The main concern being : is it logged in sentry ? (I guess you tested that the redirection was working)

Reply:
The redirect should interrupt exveption throwing